keras_cv/models/xception.py

```diff
@@ -0,0 +1,342 @@
+            "weights file to be loaded. Weights file not found at location: {weights}"
+        )
+
+    if include_top and not classes:
+        raise ValueError(
+            "If `include_top` is True, you should specify `classes`. "
+            f"Received: classes={classes}"
+        )
+
+    if include_top and pooling:
+        raise ValueError(
+            f"`pooling` must be `None` when `include_top=True`."
+            f"Received pooling={pooling} and include_top={include_top}. "
+        )
+
+    # Determine proper input shape
+    img_input = utils.parse_model_inputs(input_shape, input_tensor)
+
+    x = img_input
+
+    if include_rescaling:
+        x = layers.Rescaling(scale=1 / 255.0)(x)
+
+    # entry flow
+    x = BaseBlock(
+        entry_flow_filters[0],
+        strides=(2, 2),
+        padding="valid",
+        activation_loc="last",
+        separable=False,
+        name="block1_conv1",
+    )(x)
+    x = BaseBlock(
+        entry_flow_filters[1],
+        padding="valid",
+        activation_loc="last",
+        separable=False,
+        name="block1_conv2",
+    )(x)
+
+    for i, filters in enumerate(entry_flow_filters[2:]):
+        x = XceptionStack(filters, skip_first_act=i == 0, name=f"entry_block_{i}")(x)
+
+    # middle flow
+    for i, filters in enumerate(middle_flow_filters):
+        x = ResidualBlock(filters, name=f"middle_block_{i}")(x)
+
+    for i, filters in enumerate(exit_flow_filters[:-2]):
+        x = XceptionStack(filters, name=f"exit_block_{i}")(x)
+
+    x = BaseBlock(
+        exit_flow_filters[-2], (3, 3), activation_loc="last", name="block14_sepconv1"
+    )(x)
+    x = BaseBlock(
+        exit_flow_filters[-1], (3, 3), activation_loc="last", name="block14_sepconv2"
+    )(x)
+
+    if include_top:
+        x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        x = layers.Dense(classes, activation=classifier_activation, name="predictions")(
+            x
+        )
+    else:
+        if pooling == "avg":
+            x = layers.GlobalAveragePooling2D(name="avg_pool")(x)
+        elif pooling == "max":
+            x = layers.GlobalMaxPooling2D(name="max_pool")(x)
+
+    inputs = img_input
+
+    model = tf.keras.Model(inputs, x, name=name, **kwargs)
+
+    if weights is not None:
+        model.load_weights(weights)
+
+    return model
+
+
+def Xception(
+    include_rescaling,
+    include_top,
+    name="Xception",
+    weights=None,
+    input_shape=(None, None, 3),
+    input_tensor=None,
+    pooling=None,
+    classes=None,
+    classifier_activation="softmax",
+    **kwargs,
+):
+    return GeneralXception(
+        MODEL_CONFIGS["Xception"]["entry_flow_filters"],
+        MODEL_CONFIGS["Xception"]["middle_flow_filters"],
+        MODEL_CONFIGS["Xception"]["exit_flow_filters"],
+        include_rescaling,
+        include_top,
+        name=name,
+        weights=parse_weights(weights, include_top, "xception"),
```

Comment:
wouldn't the parse_weights error out given this PR doesn't provide any weights?

Reply:
At the moment it can handle two cases 

- `weights=None` -> default random initialization
- `weights="some_local_file.h5"` -> load the weights from a local file

It gives an error if you try `weights="imagenet"`, since the corresponding file is not defined/available